### PR TITLE
Add Gauss-Lobatto quadrature

### PR DESCRIPTION
**Pull request by nicoguaro**
-Formatted code according to PEP8 (except for hyperlinks that are too long
for that).

### CODE REVIEW

**Review by certik — APPROVED** [review]
This looks good, I verified that the points and weights are correct.

**Comment by nschloe** [comment]
Explicit root-finding in polynomials is probably one of the worst things you can do for finding Lobatto points. There's a whole body of literature on how to do it better. See, e.g., [Gautschi2000](https://www.cs.purdue.edu/homes/wxg/Lobatto.ps) or Golub's classical eigenvalue approach. State of the art is probably [Laurie2001](http://www.sciencedirect.com/science/article/pii/S0377042700005069).

**Comment by certik** [comment]
I don't understand the error (https://travis-ci.org/sympy/sympy/jobs/167750991):
```
reading sources... [ 26%] modules/integrals/integrals                           

Exception occurred:

  File "ext/docscrape.py", line 222, in parse_item_name

    raise ValueError("%s is not a item name" % text)

ValueError: \ is not a item name

The full traceback has been saved in /tmp/sphinx-err-iauq8wt0.log, if you want to report the issue to the developers.

Please also report this if it was a user error, so that a better error message can be provided next time.

A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

make: *** [html] Error 1
```

**Comment by nicoguaro** [comment]
Hello @nschloe, I was following the same procedure use in the other functions. Do you think that we should focus on implementing those?

**Comment by certik** [comment]
@nschloe I created a new issue #11782 for this. Let's move the discussion about better algorithms there.

This PR is using the same algorithms that are already in SymPy, so let's finish it up and merge it. Then we can submit new PRs with better algorithms.

**Comment by nschloe** [comment]
> This PR is using the same algorithms that are already in SymPy, so let's finish it up and merge it.

Absolutely.

File: sympy/integrals/quadrature.py [hunk 1]
```diff
@@ -486,14 +509,89 @@ def gauss_jacobi(n, alpha, beta, n_digits):
...
+    >>> x, w = gauss_lobatto(4, 5)
+    >>> x
+    [-1, -0.44721, 0.44721, 1]
+    >>> w
+    [0.16667, 0.83333, 0.83333, 0.16667]
+
+    See Also
+    ========
+
+    gauss_legendre,gauss_laguerre, gauss_gen_laguerre, gauss_hermite,\
```
Comment (certik) [inline sympy/integrals/quadrature.py]: I wonder if the `\` at the end here is causing the error: https://github.com/sympy/sympy/pull/11728#issuecomment-256770695

**Comment by certik** [comment]
Try it in a new commit, see if tests pass. Then we can think how to best
fix it. Maybe it's something else.

**Comment by certik** [comment]
@nicoguaro thanks for fixing it, that was it.

**Pull request merged**